src/Resolver/Dependencies.php

```diff
@@ -24,21 +24,42 @@ public function resolve(PatchCollection $collection): void
             return;
         }
 
-        $this->io->write('  - <info>Resolving patches from dependencies.</info>');
-
+        // Using both config keys, if $allowed_dependencies is set, it takes precedence
+        $allowed_dependencies = $this->plugin->getConfig('allow-dependency-patches');
         $ignored_dependencies = $this->plugin->getConfig('ignore-dependency-patches');
 
+        // First check, if we do allow dependency patches at all.
+        if ($allowed_dependencies === []) {
+            $this->io->write('  - <info>No patches from dependencies are allowed.</info>');
+            return;
```

Comment:
I wonder if it would be more appropriate to error, if a patch from a dependency is not allowed.
Without applying the specified patches, a dependency might not work as expected.